model/registry/user.yaml

```
@@ -22,14 +22,28 @@ groups:
         brief: >
           Unique user hash to correlate information for a user in anonymized form.
         note: >
-          Useful if `user.id` or `user.name` contain confidential information and cannot be used.
+          Useful if `user.authentication.id` or `user.name` contain confidential information and cannot be used.
         examples: ['364fc68eaf4c8acec74a4e52d7d1feaa']
       - id: user.id
         type: string
         stability: experimental
         brief: >
-          Unique identifier of the user.
+          Identifies a user interacting with a system regardless of user authentication status. This identifier may be unique only through best-effort means.
+        note: >
+          The `user.id`, when populated, is expected to be generated before user is authenticated and SHOULD NOT change after the user logs in.
+          In browser scenarios `user.id` is usually stored in cookies.
+
+          It's NOT RECOMMENDED to populate this attribute when unauthenticated users are not tracked or identified by the system.
+
+          It can be a random guid or a hash of the user's IP address. This is different from `user.hash` which is a hash of a known `user.id` or `user.name`.
```

sorry, one more minor suggestion to rearrange things slightly (feel free to rephrase or not take it at all)

```suggestion
          The `user.id`, when populated, is expected to be generated before user is authenticated and SHOULD NOT change after the user logs in.

          It's NOT RECOMMENDED to populate this attribute when unauthenticated users are not tracked or identified by the system.

          In browser scenarios `user.id` is usually stored in cookies. It can also be a random guid or a hash of the user's IP address. This is different from `user.hash` which is a hash of a known `user.authentication.id` or `user.name`.
```